controllers/imageclusterinstall_controller.go

```diff
@@ -220,18 +220,28 @@ func (r *ImageClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 			return ctrl.Result{}, err
 		}
 	}
+	// This code was moved here to avoid updating the RequirementsMet condition back and forth
+	dataImage, err := r.getDataImage(ctx, bmh.Namespace, bmh.Name)
+	if err == nil && !dataImage.ObjectMeta.DeletionTimestamp.IsZero() {
+		err = fmt.Errorf("dataImage %s/%s already exists but is being deleted, probably leftover from previous installation", bmh.Namespace, bmh.Name)
+		log.Warnf(err.Error())
+		if updateErr := r.setImageReadyCondition(ctx, ici, err); updateErr != nil {
+			log.WithError(updateErr).Error("failed to update ImageClusterInstall status")
+		}
+		return ctrl.Result{}, err
+	}
 
-	res, err := r.validateBMH(ctx, log, ici, bmh)
-	if err != nil || !res.IsZero() {
-		return res, err
+	if err = r.validateBMH(ctx, log, ici, bmh); err != nil {
+		// in case of failure to validate the BMH just requeue
+		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
-	if err := r.setClusterInstallMetadata(ctx, log, ici, clusterDeployment); err != nil {
+	if err = r.setClusterInstallMetadata(ctx, log, ici, clusterDeployment); err != nil {
 		log.WithError(err).Error("failed to set ImageClusterInstall data")
 		return ctrl.Result{}, err
 	}
 
-	res, err = r.writeInputData(ctx, log, ici, clusterDeployment, bmh)
+	res, err := r.writeInputData(ctx, log, ici, clusterDeployment, bmh)
 	if !res.IsZero() || err != nil {
 		if err != nil {
 			if updateErr := r.setImageReadyCondition(ctx, ici, err); updateErr != nil {
```

Comment:
If this is setting `RequirementsMet` don't we have the same issue here (and anywhere else that sets that condition after the call to `validateBMH`, like `setHostConfiguredCondition`)?